app/routes/app/containers/AppContainer.js

```diff
@@ -44,13 +46,99 @@ const mapStateToProps = state => ({
   balance: state.balance,
   payment: state.payment,
   transaction: state.transaction,
+
   form: state.form,
+  payform: state.payform,
+  requestform: state.requestform,
+
   invoice: state.invoice,
   modal: state.modal,
 
   currentTicker: tickerSelectors.currentTicker(state),
-  isOnchain: formSelectors.isOnchain(state),
-  isLn: formSelectors.isLn(state)
+  isOnchain: payFormSelectors.isOnchain(state),
+  isLn: payFormSelectors.isLn(state),
+  currentAmount: payFormSelectors.currentAmount(state),
+  inputCaption: payFormSelectors.inputCaption(state),
+  showPayLoadingScreen: payFormSelectors.showPayLoadingScreen(state)
 })
 
-export default connect(mapStateToProps, mapDispatchToProps)(App)
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
```

Comment:
@VonIobro this is kind of a big PR but wanted you to see the updates to this container and hear your thoughts. I refactored the form completely to it's much easier to add field validation and error handling. One of the big changes is removing the "hackathon" style logic in the components out into the parent container/selectors